app/src/main/java/com/asfoundation/wallet/ui/iab/IabActivity.kt

```diff
@@ -258,9 +234,7 @@ class IabActivity : BaseActivity(), IabView, UriNavigator {
 
   override fun onSaveInstanceState(outState: Bundle) {
     super.onSaveInstanceState(outState)
-
-    outState.putBundle(SKU_DETAILS, skuDetails)
-    outState.putBoolean(FIRST_IMPRESSION, firstImpression)
+    presenter.onSaveInstance(outState)
```

Comment:
Does it make sense that presenter knows about the Bundle? Doesn't it violate Clean Architecture's Dependency Rule?